ui/js/models/repository.js

```diff
@@ -53,22 +54,33 @@ treeherder.factory('ThRepositoryModel', [
          * We want to add this repo as watched, but we also
          * want to get the treestatus for it
          */
-        var watchRepo = function(repoName) {
-            _.extend(repos[repoName], {
+        var watchRepo = function(name) {
+            // Safeguard: Don't allow duplicates in the watch list
```

Comment:
Safeguard because Angular bombs on iterating through duplicate values in a an array.